cmake/compileroptions.cmake

```diff
@@ -82,6 +82,11 @@ elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
    add_compile_options_safe(-Wno-suggest-override) # TODO: enable when warnings are fixed in in tinyxml2
    add_compile_options_safe(-Wno-suggest-destructor-override) # TODO: enable when warnings are fixed in in tinyxml2
    add_compile_options_safe(-Wno-extra-semi-stmt) # TODO: enable when warnings are fixed in in tinyxml2
+   add_compile_options_safe(-Wno-implicitly-unsigned-literal)
+   add_compile_options_safe(-Wno-tautological-type-limit-compare)
+   add_compile_options_safe(-Wno-unused-member-function)
+   add_compile_options_safe(-Wno-deprecated-declarations)
```

Comment:
This should not be disabled since it shows us deprecated Qt functions. We need to disable it for the readdir_r case only.

Reply:
can it be disabled for readdir_r only?